### PR TITLE
[SIL Opaque Value] Add support for open_existential_box_value in AddressLowering

### PR DESCRIPTION
**Pull request by meg-gupta**

### CODE REVIEW

File: lib/SILOptimizer/Mandatory/AddressLowering.cpp [hunk 1]
```diff
@@ -2978,6 +2971,14 @@ class DefRewriter : SILInstructionVisitor<DefRewriter> {
     for (Operand &operand : tupleInst->getAllOperands())
       addrMat.initializeComposingUse(&operand);
   }
+
+  void markRewritten(SILValue oldValue, SILValue addr) {
```
Comment (atrick) [inline lib/SILOptimizer/Mandatory/AddressLowering.cpp:2975]: This doesn't actually mark the storage rewritten. So maybe it should be called something else. Like `setStorageAddress`

File: lib/SILOptimizer/Mandatory/AddressLowering.cpp [hunk 2]
```diff
@@ -2939,11 +2925,18 @@ class DefRewriter : SILInstructionVisitor<DefRewriter> {
     addrMat.initializeComposingUse(&initExistentialValue->getOperandRef());
   }
 
-  // Project an opaque value out of a box-type existential.
   void visitOpenExistentialBoxValueInst(
-      OpenExistentialBoxValueInst *openExistentialBox) {
-    // FIXME: Unimplemented
-    llvm::report_fatal_error("Unimplemented OpenExistentialBoxValue def.");
+      OpenExistentialBoxValueInst *openExistentialBoxValue) {
```
Comment (atrick) [inline lib/SILOptimizer/Mandatory/AddressLowering.cpp:2929]: Is the operand of open_existential_box_value always a loadable type (a box existential)? If not, then it also needs to be handled in the use rewriter.... just double checking out of paranoia.

File: test/SILOptimizer/address_lowering_misc.sil [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// RUN: %target-sil-opt -address-lowering -enable-sil-opaque-values -sil-verify-all %s | %FileCheck %s
```
Comment (atrick) [inline test/SILOptimizer/address_lowering_misc.sil:1]: I used the `_lib` suffix for tests that need to import the stdlib. as in `test/SILGen/opaque_values_silgen_lib.swift`

**Review by atrick — APPROVED** [review]
Cool, thanks. A few minor comments.

File: lib/SILOptimizer/Mandatory/AddressLowering.cpp (hunk 2 above)
Comment (meg-gupta) [reply lib/SILOptimizer/Mandatory/AddressLowering.cpp:2929]: Looking at the verifier, I think this is the case -  https://github.com/apple/swift/blob/66b08a8a6cce969f1e3e90280b7b40d79cc056eb/lib/SIL/Verifier/SILVerifier.cpp#L3703

**Pull request merged**